### PR TITLE
Add support for loading the raw json template file

### PR DESCRIPTION
**Pull request by ruflin**
Current in 6.x templates can only be loaded from the fields.yml and are generated or have to be loaded manually. To allow users to create their own template.json file and still use the Beat to load it, `setup.template.raw.path` can now be used.

### CODE REVIEW

**Comment by ruflin** [comment]
@tsg Any input on the config naming?

**Comment by tsg** [comment]
Maybe `setup.template.json.path`? or `raw_json`?

**Comment by tsg** [comment]
(pressed the wrong button, sorry)

File: winlogbeat/docs/fields.asciidoc [hunk 1]
```diff
@@ -215,7 +215,7 @@ Contains common fields available in all event types.
 required: True
 
 The event log API type used to read the record. The possible values are "wineventlog" for the Windows Event Log API or "eventlogging" for the Event Logging API.
-The Event Logging API was designed for Windows Server 2003 or Windows 2000 operating systems. In Windows Vista, the event logging infrastructure was redesigned. On Windows Vista or later operating systems, the Windows Event Log API is used. Winlogbeat automatically detects which API to use for reading event logs.
+The Event Logging API was designed for Windows Server 2003, Windows XP, or Windows 2000 operating systems. In Windows Vista, the event logging infrastructure was redesigned. On Windows Vista or later operating systems, the Windows Event Log API is used. Winlogbeat automatically detects which API to use for reading event logs.
```
Comment (andrewkroh) [inline winlogbeat/docs/fields.asciidoc]: That sounds correct, but doesn't seem related to this PR.

**Comment by ruflin** [comment]
The config option is now called `setup.template.json`. Docs and tests were added. Ready for review.

**Comment by ruflin** [comment]
CI failure is not related.
